sizefashion/src/reducers/filter_reducer.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  LOAD_PRODUCTS,
+  UPDATE_FILTERS,
+  FILTER_PRODUCTS,
+  CLEAR_FILTERS,
+} from '../actions'
+import { initialStateType } from '../context/filter_context'
+
+const filter_reducer = ( state: initialStateType, action: { type: any; payload?: any } ) => {
+  if (action.type === LOAD_PRODUCTS) {
+    return {
+      ...state,
+      allProducts: [...action.payload],
+      filteredProducts: [...action.payload],
+      filters: { ...state.filters },
+    }
+  }
+  if (action.type === UPDATE_FILTERS) {
+    let { name, value } = action.payload
+    return { ...state, filters: { ...state.filters, [name]: value } }
+  }
+  if (action.type === FILTER_PRODUCTS) {
+    const { allProducts } = state
+    const {
+      searchTerm,
+      category,
+      vendor,
+    } = state.filters
+
+    let tempProducts = [...allProducts]
+    
+    if (searchTerm) {
+      tempProducts = tempProducts.filter(product => {
+        return (
+          product.title.toLowerCase().includes(searchTerm.toLowerCase())
+        )
+      })
+    }
+    // category
+    if (category !== 'all') {
+      tempProducts = tempProducts.filter(product => {
+        return product.category === category
+      })
+    }
+
+    if (vendor !== 'all') {
+      tempProducts = tempProducts.filter(product => {
+        return product.vendor === vendor
+      })
+    }
```

Comment:
@otaaviostein when creating a reducer, it's a good practice to opt for a `switch case` strcuture, because it controls your condition flow better. Creating a if chain condition can make your code readability a bit difficult to go with.